src/units/animation_component.cpp

```diff
@@ -209,17 +205,12 @@ void unit_animation_component::reset_affect_adjacent(const unit_map& units)
 		}
 	}
 	if(affect_adjacent) {
-		const auto adjacent = get_adjacent_tiles(u_.get_location());
-		for(unsigned i = 0; i < adjacent.size(); ++i) {
-			const unit_map::const_iterator it = units.find(adjacent[i]);
```

Comment:
Checking 'units' once per hex and reviewing the entire variable if an adjacent hex is empty seems more expensive than what I'm proposing.

Reply:
Do you mean find is iterating all units? That would be very bad.

Reply:
https://github.com/wesnoth/wesnoth/blob/5e819a3bdf35c69f0251c5fe9937b5c4638b0187/src/units/map.hpp#L40

Reply:
units.find() means that it is looking for a unit in 'units' whose position matches (if the unit exists), and therefore it must go through the units, and I don't see how it would do otherwise

Reply:
> https://github.com/wesnoth/wesnoth/blob/5e819a3bdf35c69f0251c5fe9937b5c4638b0187/src/units/map.hpp#L40

what should i understand? what find * 6 remain lees expensive what that i propose?

Reply:
It says [O(1)](https://en.wikipedia.org/wiki/Big_O_notation) so that's constant time regardless of the number of units. The unit map contains a location -> unit mapping so it's a direct map (std::unordered_map) access. There would have to be very few units in total for it to be faster to go through all units instead.